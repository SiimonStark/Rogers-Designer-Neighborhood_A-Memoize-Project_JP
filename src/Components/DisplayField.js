import React, { Component } from 'react';
// import BurgerMenu from './BurgerMenu';

class DisplayField extends Component {
  constructor() {
    super()
    this.state = {
      gridSquares: ['square-1','square-2','square-3','square-4'],
      gridContainer: true
    }
  }

  render() {
    const fieldStyle = { display: 'grid', border: '1px solid red', gridTemplateAreas: `"square-1 square-2" "square-3 square-4"` };
    return (
      <main className="DisplayField">
        <div className="grass"></div>
        {  this.state.gridContainer ?
            <section className="field-container" style={fieldStyle}>
            {this.state.gridSquares.map(square => <div className={square}>{square}</div>)}
            </section> :
            <section className="field-container">
              {this.state.gridSquares.map(square => <div className={square}>{square}</div>)}
            </section>
        }
      </main>
    );
  }
}

export default DisplayField;