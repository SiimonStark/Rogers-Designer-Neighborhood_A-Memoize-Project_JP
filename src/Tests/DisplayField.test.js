import React from 'react';
import DisplayField from '../Components/DisplayField';
import { shallow } from 'enzyme';

let mockState = {
  gridContainer: true,
  gridSquares: [
    "fieldSquare square-1",
    "fieldSquare square-2",
    "fieldSquare square-3",
    "fieldSquare square-4"
  ],
  items: {
    road: "../Resources/town/road.png",
    trees: [
      "../Resources/town/tree-2.png",
      "../Resources/town/tree-3.png",
      "../Resources/town/tree-4.png"
    ],
    house: "../Resources/town/house-1.png"
  }
};

describe('DisplayField', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <DisplayField />
      );
  });

  it('should match snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(mockState);
  })
});