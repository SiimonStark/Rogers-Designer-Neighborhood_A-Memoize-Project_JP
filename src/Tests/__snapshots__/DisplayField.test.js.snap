// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayField should match snapShot 1`] = `
<main
  className="DisplayField"
>
  <div
    className="grass"
  />
  <div
    className="road1--container"
  >
    <div
      className="road1 positionRoad"
    />
  </div>
  <div
    className="road2--container"
  >
    <div
      className="road2 positionRoad"
    />
  </div>
  <section
    className="field-container"
    style={
      Object {
        "display": "grid",
        "gridTemplateAreas": "\\"square-1 square-2\\" \\"square-3 square-4\\"",
      }
    }
  >
    <div
      className="fieldSquare square-1"
      key="0"
    >
      <p>
        S: 1
      </p>
      <div
        className="road--container"
      >
        <div
          className="road positionRoad"
        />
      </div>
      <div
        className="tree--container"
      >
        <div
          className="tree1 positionTree1"
        />
        <div
          className="tree2 positionTree2"
        />
        <div
          className="tree3 positionTree3"
        />
      </div>
      <div
        className="house--container"
      >
        <div
          className="house positionHouse"
        />
      </div>
    </div>
    <div
      className="fieldSquare square-2"
      key="1"
    >
      <p>
        S: 2
      </p>
      <div
        className="road--container"
      >
        <div
          className="road positionRoad"
        />
      </div>
      <div
        className="tree--container"
      >
        <div
          className="tree1 positionTree1"
        />
        <div
          className="tree2 positionTree2"
        />
        <div
          className="tree3 positionTree3"
        />
      </div>
      <div
        className="house--container"
      >
        <div
          className="house positionHouse"
        />
      </div>
    </div>
    <div
      className="fieldSquare square-3"
      key="2"
    >
      <p>
        S: 3
      </p>
      <div
        className="road--container"
      >
        <div
          className="road positionRoad"
        />
      </div>
      <div
        className="tree--container"
      >
        <div
          className="tree1 positionTree1"
        />
        <div
          className="tree2 positionTree2"
        />
        <div
          className="tree3 positionTree3"
        />
      </div>
      <div
        className="house--container"
      >
        <div
          className="house positionHouse"
        />
      </div>
    </div>
    <div
      className="fieldSquare square-4"
      key="3"
    >
      <p>
        S: 4
      </p>
      <div
        className="road--container"
      >
        <div
          className="road positionRoad"
        />
      </div>
      <div
        className="tree--container"
      >
        <div
          className="tree1 positionTree1"
        />
        <div
          className="tree2 positionTree2"
        />
        <div
          className="tree3 positionTree3"
        />
      </div>
      <div
        className="house--container"
      >
        <div
          className="house positionHouse"
        />
      </div>
    </div>
  </section>
</main>
`;
